xc/xenops_server_xen.ml

```diff
@@ -540,18 +540,25 @@ let device_by_id xc xs vm kind domain_selection id =
     raise (Does_not_exist("domain", vm))
   | Some frontend_domid ->
     let fetch_all_from_xenstore_f () = (* expensive *)
-      let devices = Device_common.list_frontends ~xs frontend_domid in
+      let devices = (Device_common.list_frontends ~xs frontend_domid)
+        @ (Device_common.list_xenserver_devices ~xs frontend_domid)
+      in
       let key = _device_id kind in
       let id_of_device device =
-        let path = Device_common.get_private_data_path_of_device device in
```

Comment:
I don't think we should need to change the `private` data path for SR-IOV devices. This is a namespace that is ours, so we don't have to make the distinction between "upstream" device kinds (vifs, vbds) and our own invented ones (for SR-IOV).

If we don't have the private paths yet for SR-IOV devices, then we should add them.

Reply:
This comment makes me feel relieved.
Making the distinction between "upstream" device kinds makes lots of things different in Xenopsd.